test/functional/shada/history_spec.lua

```diff
@@ -173,158 +171,57 @@ describe('ShaDa support code', function()
-  function()
-    set_additional_cmd('set encoding=latin1')
-    reset()
-    -- \171 is U+00AB LEFT-POINTING DOUBLE ANGLE QUOTATION MARK in latin1
-    nvim_feed(':echo "\171"\n')
-    set_additional_cmd('')
-    nvim_command('qall')
-    reset()
-    eq('echo "«"', funcs.histget(':', -1))
-  end)
-
-  it('dumps and loads history correctly when &encoding /= UTF-8 when loading',
-  function()
-    -- \171 is U+00AB LEFT-POINTING DOUBLE ANGLE QUOTATION MARK in latin1
     nvim_feed(':echo "«"\n')
-    set_additional_cmd('set encoding=latin1')
     nvim_command('qall')
     reset()
-    eq('echo "\171"', funcs.histget(':', -1))
+    eq('echo "«"', funcs.histget(':', -1))
   end)
 
-  it('dumps and loads replacement correctly when &encoding is not UTF-8',
+  it('dumps and loads replacement with UTF-8 characters',
   function()
-    set_additional_cmd('set encoding=latin1')
-    reset()
-    -- \171 is U+00AB LEFT-POINTING DOUBLE ANGLE QUOTATION MARK in latin1
-    nvim_command('substitute/./\171/ge')
+    nvim_command('substitute/./«/ge')
     nvim_command('qall!')
     reset()
     funcs.setline('.', {'.'})
     nvim_command('&')
-    eq('\171', funcs.getline('.'))
-  end)
-
-  it('dumps&loads replacement correctly when &encoding /= UTF-8 when dumping',
-  function()
-    set_additional_cmd('set encoding=latin1')
-    reset()
-    -- \171 is U+00AB LEFT-POINTING DOUBLE ANGLE QUOTATION MARK in latin1
-    nvim_command('substitute/./\171/ge')
-    set_additional_cmd('')
-    nvim_command('qall')
-    reset()
-    funcs.setline('.', {'.'})
-    nvim_command('&')
     eq('«', funcs.getline('.'))
   end)
 
-  it('dumps&loads replacement correctly when &encoding /= UTF-8 when loading',
-  function()
-    -- \171 is U+00AB LEFT-POINTING DOUBLE ANGLE QUOTATION MARK in latin1
-    nvim_command('substitute/./«/ge')
-    set_additional_cmd('set encoding=latin1')
-    nvim_command('qall')
-    reset()
-    funcs.setline('.', {'.'})
-    nvim_command('&')
-    eq('\171', funcs.getline('.'))
-  end)
-
-  it('dumps and loads substitute pattern correctly when &encoding is not UTF-8',
+  it('dumps and loads substitute pattern with UTF-8 characters',
   function()
-    set_additional_cmd('set encoding=latin1')
-    reset()
-    -- \171 is U+00AB LEFT-POINTING DOUBLE ANGLE QUOTATION MARK in latin1
-    nvim_command('substitute/\171/./ge')
+    nvim_command('substitute/«/./ge')
     nvim_command('qall!')
     reset()
-    funcs.setline('.', {'\171«'})
-    nvim_command('&')
-    eq('.«', funcs.getline('.'))
-  end)
-
-  it('dumps&loads s/pattern correctly when &encoding /= UTF-8 when dumping',
-  function()
-    set_additional_cmd('set encoding=latin1')
-    reset()
-    -- \171 is U+00AB LEFT-POINTING DOUBLE ANGLE QUOTATION MARK in latin1
-    nvim_command('substitute/\171/./ge')
-    set_additional_cmd('')
-    nvim_command('qall')
-    reset()
     funcs.setline('.', {'«\171'})
     nvim_command('&')
+    --nvim_command('substitute/«/./ge')
```

Comment:
`:&` does not repeat flags so the comment should be like

    nvim_command('&')  -- :substitute/«/./